api/MediaSource.json

```diff
@@ -663,50 +670,113 @@
-            },
-            "chrome_android": {
-              "version_added": "25"
-            },
+            "chrome": [
+              {
+                "version_added": "53"
+              },
+              {
+                "version_added": "31",
+                "version_removed": "53",
+                "partial_implementation": true,
+                "notes": "The <code>onsourceclose</code> event handler property is not supported."
+              }
+            ],
+            "chrome_android": [
+              {
+                "version_added": "53"
+              },
+              {
+                "version_added": "31",
+                "version_removed": "53",
+                "partial_implementation": true,
+                "notes": "The <code>onsourceclose</code> event handler property is not supported."
+              }
+            ],
             "edge": {
-              "version_added": "12"
-            },
-            "firefox": {
-              "version_added": "42"
-            },
-            "firefox_android": {
-              "version_added": "41"
+              "version_added": "17"
             },
+            "firefox": [
+              {
+                "version_added": "87"
+              },
+              {
+                "version_added": "42",
+                "partial_implementation": true,
+                "notes": "The <code>onsourceclose</code> event handler property is not supported."
+              }
+            ],
+            "firefox_android": [
+              {
+                "version_added": "87"
+              },
+              {
+                "version_added": "42",
+                "partial_implementation": true,
+                "notes": "The <code>onsourceclose</code> event handler property is not supported."
+              }
+            ],
             "ie": {
               "version_added": "11",
               "notes": "Only works on Windows 8+."
             },
-            "opera": {
-              "version_added": "15"
-            },
-            "opera_android": {
-              "version_added": "14"
-            },
+            "opera": [
+              {
+                "version_added": "40"
+              },
+              {
+                "version_added": "18",
+                "version_removed": "40",
+                "partial_implementation": true,
+                "notes": "The <code>onsourceclose</code> event handler property is not supported."
+              }
+            ],
+            "opera_android": [
+              {
+                "version_added": "41"
+              },
+              {
+                "version_added": "18",
+                "version_removed": "41",
+                "partial_implementation": true,
+                "notes": "The <code>onsourceclose</code> event handler property is not supported."
+              }
+            ],
             "safari": {
-              "version_added": "8"
+              "version_added": "10.1"
```

Comment:
I'm pretty sure Safari also had the events before the event handler properties. This will need researching. The original data saying Safari 8 here probably has some truth to it.

Reply:
Let me know if you do the research now or if that's a follow-up.

Reply:
When a PR changes a version number I consider the research to back it up required, but reverting Safari changes and leaving it for a follow-up would have been fine.

Here's the research steps I just did:

- Go to https://github.com/WebKit/WebKit/find/main
- Type "MediaSource.idl" to find https://github.com/WebKit/WebKit/blob/main/Source/WebCore/Modules/mediasource/MediaSource.idl
- Blame to see https://github.com/WebKit/WebKit/blame/main/Source/WebCore/Modules/mediasource/MediaSource.idl
- Look for the on* entries and what commit introduced them

That leads to https://github.com/WebKit/WebKit/commit/3fc9d2586de516be3f454e84733a74c76e2e81ff, so we can say that all of the on* event handler properties should have the same version as `setLiveSeekableRange` and `clearLiveSeekableRange`, and also that the events were supported earlier.

Using the instructions in https://developer.mozilla.org/en-US/docs/MDN/Contribute/Processes/Matching_features_to_browser_version#safari we can figure out what the likely version number is:
https://trac.webkit.org/browser/webkit/trunk/Source/WebCore/Configurations/Version.xcconfig?rev=206146

603.1.6 maps to Safari 10.1, which does match `setLiveSeekableRange`.

So, let's mark the events as partially supported since Safari 8, and fully since Safari 10.1. I'll send a PR to fix the incorrect `clearLiveSeekableRange` data.

Reply:
I've sent https://github.com/mdn/browser-compat-data/pull/15064.

Reply:
Thanks for the detailed break down. I think we should probably have some BCD docs per browser that outlines how to do this sort of research. The MDN page is somewhat useful, though. 

I added a commit to fix Safari data